processing/src/main/java/io/druid/segment/FilteredOffset.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import io.druid.collections.bitmap.ImmutableBitmap;
+import io.druid.query.BaseQuery;
+import io.druid.query.filter.BooleanFilter;
+import io.druid.query.filter.Filter;
+import io.druid.query.filter.RowOffsetMatcherFactory;
+import io.druid.query.filter.ValueMatcher;
+import io.druid.query.monomorphicprocessing.RuntimeShapeInspector;
+import io.druid.segment.data.Offset;
+import io.druid.segment.filter.BooleanValueMatcher;
+import io.druid.segment.historical.HistoricalCursor;
+import io.druid.segment.historical.OffsetHolder;
+import org.roaringbitmap.IntIterator;
+
+final class FilteredOffset extends Offset
```

Comment:
This class if factored out of `QueryableIndexStorageAdapter`. It didn't exist before this PR, because `QueryableIndexStorageAdapter` used to handle post filters using a special cursor, and now there is the same cursor accepting any provided `Offset`, post-filtered like `FilteredOffset`, or not. So this class is tested by all the same tests which test `QueryableIndexStorageAdapter` directly or indirectly, mostly tests of specific query types.